Providers/Profile.swift

```diff
@@ -949,8 +949,16 @@ open class BrowserProfile: Profile {
 
         fileprivate func syncClientsWithDelegate(_ delegate: SyncDelegate, prefs: Prefs, ready: Ready, why: SyncReason) -> SyncResult {
             log.debug("Syncing clients to storage.")
+
+            func notifyAll() {
+                guard let account = self.profile.getAccount() else {
+                    return
+                }
+                account.notifyAll(collectionsChanged: ["clients"])
+            }
+
             let clientSynchronizer = ready.synchronizer(ClientsSynchronizer.self, delegate: delegate, prefs: prefs, why: why)
-            return clientSynchronizer.synchronizeLocalClients(self.profile.remoteClientsAndTabs, withServer: ready.client, info: ready.info) >>== { result in
+            return clientSynchronizer.synchronizeLocalClients(self.profile.remoteClientsAndTabs, withServer: ready.client, info: ready.info, onFirstSync: notifyAll) >>== { result in
```

Comment:
_Or_, if you define this behavior as being an intrinsic part of the clients collection, define a push recipient protocol, and supply an instance to the synchronizer to use as it sees fit. Then you would put those side effects inside the synchronizer.